### PR TITLE
renamed prev_state_root in DoBlockRequest to prev_block_hash

### PR DESCRIPTION
**Pull request by akansjain**
In crates/astria-proto/proto/astria/execution/v1/execution.proto, there is a the gRPC method DoBlockRequest has a field called prev_state_root but what's actually passed here is the parent block hash, and should be called prev_block_hash.

Therefore, field prev_state_root is renamed to prev_block_hash in DoBlockRequest.

### CODE REVIEW

**Comment by akansjain** [comment]
Hey @noot ,
Please review my PR.

**Comment by joroshiba** [comment]
Thanks for the contribution, can you also update `astria-conductor` to use the new grpc as well [here](https://github.com/astriaorg/astria/blob/main/crates/astria-conductor/src/execution_client.rs#LL68C13-L68C28)

File: crates/astria-conductor/src/execution_client.rs [hunk 1]
```diff
@@ -64,8 +64,8 @@ impl ExecutionClient for ExecutionRpcClient {
         timestamp: Option<Timestamp>,
     ) -> Result<DoBlockResponse> {
         let request = DoBlockRequest {
-            // TODO: this field name should actually be prev_block_hash!
-            prev_state_root: prev_block_hash,
+            // Updated field name: prev_block_hash
```
Comment (joroshiba) [inline crates/astria-conductor/src/execution_client.rs:67]: We can just remove this comment.

File: crates/astria-conductor/src/execution_client.rs [hunk 2]
```diff
@@ -64,8 +64,8 @@ impl ExecutionClient for ExecutionRpcClient {
...
+            prev_block_hash: prev_block_hash.clone(),
```
Comment (joroshiba) [inline crates/astria-conductor/src/execution_client.rs:68]: I don't think the clone is necessary here, would prefer the shorthand.

Comment (noot) [inline crates/astria-conductor/src/execution_client.rs:68]:
```suggestion
            prev_block_hash,
```

File: crates/astria-conductor/src/execution_client.rs (hunk 1 above)
Comment (joroshiba) [inline crates/astria-conductor/src/execution_client.rs:67]:
```suggestion
```

**Review by joroshiba — CHANGES_REQUESTED** [review]

**Comment by noot** [comment]
closing due to staleness

**Pull request closed without merging**